### ISSUE TITLE
IndexOutOfRange in ComputeHeapGraphData

### ISSUE THREAD

**Issue by KirillOsenkov** [KirillOsenkov/MSBuildStructuredLog#637]
```
System.IndexOutOfRangeException: Index was outside the bounds of the array. 
   at StructuredLogViewer.Controls.TracingControl.ComputeHeatGraphData(Double unitDuration) in C:\MSBuildStructuredLog\src\StructuredLogViewer\Controls\TracingControl.xaml.cs:line 368 
   at StructuredLogViewer.Controls.TracingControl.CreateActivityLineGraph() in C:\MSBuildStructuredLog\src\StructuredLogViewer\Controls\TracingControl.xaml.cs:line 497 
   at StructuredLogViewer.Controls.TracingControl.Draw() in C:\MSBuildStructuredLog\src\StructuredLogViewer\Controls\TracingControl.xaml.cs:line 455 
   at StructuredLogViewer.Controls.TracingControl.ComputeAndDraw() in C:\MSBuildStructuredLog\src\StructuredLogViewer\Controls\TracingControl.xaml.cs:line 301 
   at StructuredLogViewer.Controls.TracingControl.SetTimeline(Timeline timeline, Int64 globalStart, Int64 globalEnd) in C:\MSBuildStructuredLog\src\StructuredLogViewer\Controls\TracingControl.xaml.cs:line 289 
   at StructuredLogViewer.Controls.BuildControl.PopulateTrace() in C:\MSBuildStructuredLog\src\StructuredLogViewer\Controls\BuildControl.xaml.cs:line 412 
   at StructuredLogViewer.Controls.BuildControl.CentralTabControl_SelectionChanged(Object sender, SelectionChangedEventArgs e) in C:\MSBuildStructuredLog\src\StructuredLogViewer\Controls\BuildControl.xaml.cs:line 322 
```

FYI @yuehuang010

**Comment by KirillOsenkov** [comment]
Let me know if you want a sample binlog for this, I can email it to you.

**Comment by yuehuang010** [comment]
Could you please add a binlog.  If you zip the binlog you can host it on github.

**Comment by KirillOsenkov** [comment]
There are dangers associated with publishing binlogs publicly. It has environment variables, customer's secrets potentially, etc. so the guidance is to avoid sharing binlogs publicly unless it's clear that it has no secrets.

**Comment by KirillOsenkov** [comment]
Sent the binlog to you on Teams.

**Comment by KirillOsenkov** [comment]
Fixed in #638
